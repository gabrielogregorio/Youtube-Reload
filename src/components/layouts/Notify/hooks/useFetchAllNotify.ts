import { useEffect, useState } from 'react';
import { NotifyService } from '@/layouts/Notify/services/NotifyService';
import { FetchNotifyService } from '@/layouts/Notify/services/FetchNotifyService';
import { NotifyFromApiMapper } from '@/layouts/Notify/mappers/fromApi';

export const useFetchAllNotify = () => {
  const [isLoading, setIsLoading] = useState<boolean>(false);
  const [error, setError] = useState<string | undefined>(undefined);
  const [notifications, setNotifications] = useState<NotifyFromApiMapper[]>([]);
  const [viewedNotificationIds, setViewedNotificationIds] = useState<number[]>([]);

  useEffect(() => {
    setIsLoading(false);
    setError(undefined);
    setNotifications([]);

    FetchNotifyService.fetch()
      .then((res: NotifyFromApiMapper[]) => {
        setNotifications(res);
      })
      .catch(() => {
        setError('Error on Load Data');
      })
      .finally(() => {
        setIsLoading(false);
      });
  }, []);

  useEffect(() => {
    const items = NotifyService.getOrInitialize();
    setViewedNotificationIds(items);
  }, []);

  const updateViewedNotifications = (item: number[]): void => {
    const newNotifies = NotifyService.updateNotify(item);
    setViewedNotificationIds(newNotifies);
  };

  return {
    isLoading,
    error,
    notifications,
    viewedNotificationIds,
    updateViewedNotifications,
  };
};
