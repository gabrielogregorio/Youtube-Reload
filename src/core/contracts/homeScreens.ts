export enum ScreenEnum {
  'home' = '/youtube-reload',
  'all' = '/youtube-reload/all',
  'likes' = '/youtube-reload/likes',
  'unlikes' = '/youtube-reload/unlikes',
}
