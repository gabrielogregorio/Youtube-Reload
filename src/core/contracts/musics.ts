import type { CategoryEnum } from '@/data/data.reload';

export interface IMusicApi {
  year: number;
  artist: string;
  id: string;
  title: string;
  durationInSeconds: number;
  category: CategoryEnum;
  views: string;
  minuteToSkipIntroduction?: number;
  comments: string;
  likes: string;
}

export interface IMusic {
  year: number;
  artist: string;
  id: string;
  title: string;
  durationInSeconds: number;
  category: CategoryEnum;
  minuteToSkipIntroduction?: number;

  viewsAsString: string;
  views: number;

  commentsAsString: string;
  comments: number;

  likesAsString: string;
  likes: number;

  approval: number;
  approvalComments: number;

  searchStringNormalized: string;
}

interface ITransformedType extends IMusic {
  img: string;
  url: string;
}

export interface IMusicWithTransformation extends ITransformedType {}
