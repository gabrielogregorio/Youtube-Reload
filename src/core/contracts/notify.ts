export interface INotifyApi {
  id: number;
  title: string;
  date: string;
  emoji: string;
}

export interface INotify {
  id: number;
  title: string;
  date: string;
  emoji: string;
}
