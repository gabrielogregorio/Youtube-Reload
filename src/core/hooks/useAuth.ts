import { StorageAccessNameEnum, StorageService } from '@/services/StorageService';
import { LogService } from '@/services/log/LogService';
import { useEffect } from 'react';

export const useAuth = (): void => {
  useEffect(() => {
    const userId = StorageService.getItem(StorageAccessNameEnum.UserId);

    const payloadUserId = userId ? { userId } : { userId: new Date().getTime().toString() };

    LogService.setUser(payloadUserId);
  }, []);
};
