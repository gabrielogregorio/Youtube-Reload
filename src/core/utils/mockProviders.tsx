import { Provider } from 'react-redux';
import { store } from '@/connections/store';
import type { ReactElement, ReactNode } from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';

export const MockProviders = ({ children }: { children: ReactNode }): ReactElement => (
  <Provider store={store}>
    <BrowserRouter>
      <Routes>
        <Route path="*" element={children} />
      </Routes>
    </BrowserRouter>
  </Provider>
);
