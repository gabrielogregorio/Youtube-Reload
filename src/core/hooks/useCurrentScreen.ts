import { useCallback } from 'react';
import { useAppDispatch, useAppSelector } from '@/connections/store/useRedux';
import { ScreenEnum } from '@/contracts/homeScreens';
import { setScreen } from '@/connections/features/currentScreen/slices';
import { screenSelectedSelector } from '@/connections/features/currentScreen/selectors';

export const useCurrentScreen = () => {
  const dispatch = useAppDispatch();
  const { currentScreen } = useAppSelector(screenSelectedSelector);

  const updateCurrentScreen = useCallback(
    (newScreen: ScreenEnum) => {
      dispatch(setScreen(newScreen));
    },
    [dispatch],
  );

  return {
    currentScreen,
    updateCurrentScreen,
  };
};
