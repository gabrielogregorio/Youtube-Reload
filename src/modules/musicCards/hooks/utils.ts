import { MusicFromApiMapper } from '@/modules/musicCards/mappers/get/fromApi';
import { shuffleArray } from '@/utils/generators';

export const applyOffsetAndLimit = (offset: number, limit: number, filtered: MusicFromApiMapper[]) => {
  return filtered.slice(offset, limit);
};

export const generateRandomPlaylist = (filtered: MusicFromApiMapper[]) => {
  return shuffleArray(filtered);
};
