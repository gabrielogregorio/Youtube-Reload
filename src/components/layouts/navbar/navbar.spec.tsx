import { ScreenEnum } from '@/contracts/homeScreens';
import { render, screen } from '@testing-library/react';
import { vi } from 'vitest';
import { MockProviders } from '@/utils/mockProviders';
import * as useCurrentScreen from '@/hooks/useCurrentScreen';
import { Navbar } from '.';

const mockNavbar: { index: number; screen: ScreenEnum; text: string }[] = [
  {
    index: 1,
    screen: ScreenEnum.home,
    text: 'Home',
  },
  {
    index: 2,
    screen: ScreenEnum.likes,
    text: 'Likes',
  },
  {
    index: 3,
    screen: ScreenEnum.unLikes,
    text: 'UnLikes',
  },
  {
    index: 4,
    screen: ScreenEnum.all,
    text: 'All',
  },
  {
    index: 5,
    screen: ScreenEnum.configs,
    text: 'Configs',
  },
];

const spyUseAuth = vi.spyOn(useCurrentScreen, 'useCurrentScreen');

spyUseAuth.mockImplementation(() => {
  return {
    currentScreen: ScreenEnum.home,
    updateCurrentScreen: () => {},
  };
});

describe('<Navbar />', () => {
  afterEach(() => {
    vi.restoreAllMocks();
  });

  it('should render navbar', () => {
    render(
      <MockProviders>
        <Navbar />
      </MockProviders>,
    );
    expect(screen.getAllByRole('button')).toHaveLength(mockNavbar.length);
  });

  it('should render button texts', () => {
    const { container } = render(
      <MockProviders>
        <Navbar />
      </MockProviders>,
    );

    expect(container).toMatchSnapshot();
  });
});
