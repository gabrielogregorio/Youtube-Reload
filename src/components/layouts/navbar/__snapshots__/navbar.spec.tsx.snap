// Vitest Snapshot v1

exports[`<Navbar /> > should render button texts 1`] = `
<div>
  <nav
    aria-label="Main Navigation"
    class="w-full flex items-center justify-center"
    role="navigation"
  >
    <button
      aria-label="Navigate to Home"
      class="border-b-[3px] bg-transparent decoration-none p-[20px] pb-[5px] mb-[15px] cursor-pointer transition-all duration-150 hover:scale-110 select-none font-bold flex items-center justify-center border-red-light text-red-light"
      type="button"
    >
      <span
        class="md:hidden"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M946.5 505L534.6 93.4a31.93 31.93 0 0 0-45.2 0L77.5 505c-12 12-18.8 28.3-18.8 45.3 0 35.3 28.7 64 64 64h43.4V908c0 17.7 14.3 32 32 32H448V716h112v224h265.9c17.7 0 32-14.3 32-32V614.3h43.4c17 0 33.3-6.7 45.3-18.8 24.9-25 24.9-65.5-.1-90.5z"
          />
        </svg>
      </span>
      <span
        class="ml-2 hidden md:block text-[0.9rem]"
      >
        Home
      </span>
    </button>
    <button
      aria-label="Navigate to Likes"
      class="border-b-[3px] bg-transparent decoration-none p-[20px] pb-[5px] mb-[15px] cursor-pointer transition-all duration-150 hover:scale-110 select-none font-bold flex items-center justify-center border-transparent text-white"
      type="button"
    >
      <span
        class="md:hidden"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M885.9 533.7c16.8-22.2 26.1-49.4 26.1-77.7 0-44.9-25.1-87.4-65.5-111.1a67.67 67.67 0 0 0-34.3-9.3H572.4l6-122.9c1.4-29.7-9.1-57.9-29.5-79.4A106.62 106.62 0 0 0 471 99.9c-52 0-98 35-111.8 85.1l-85.9 311h-.3v428h472.3c9.2 0 18.2-1.8 26.5-5.4 47.6-20.3 78.3-66.8 78.3-118.4 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7-.2-12.6-2-25.1-5.6-37.1zM112 528v364c0 17.7 14.3 32 32 32h65V496h-65c-17.7 0-32 14.3-32 32z"
          />
        </svg>
      </span>
      <span
        class="ml-2 hidden md:block text-[0.9rem]"
      >
        Likes
      </span>
    </button>
    <button
      aria-label="Navigate to UnLikes"
      class="border-b-[3px] bg-transparent decoration-none p-[20px] pb-[5px] mb-[15px] cursor-pointer transition-all duration-150 hover:scale-110 select-none font-bold flex items-center justify-center border-transparent text-white"
      type="button"
    >
      <span
        class="md:hidden"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M885.9 490.3c3.6-12 5.4-24.4 5.4-37 0-28.3-9.3-55.5-26.1-77.7 3.6-12 5.4-24.4 5.4-37 0-28.3-9.3-55.5-26.1-77.7 3.6-12 5.4-24.4 5.4-37 0-51.6-30.7-98.1-78.3-118.4a66.1 66.1 0 0 0-26.5-5.4H273v428h.3l85.8 310.8C372.9 889 418.9 924 470.9 924c29.7 0 57.4-11.8 77.9-33.4 20.5-21.5 31-49.7 29.5-79.4l-6-122.9h239.9c12.1 0 23.9-3.2 34.3-9.3 40.4-23.5 65.5-66.1 65.5-111 0-28.3-9.3-55.5-26.1-77.7zM112 132v364c0 17.7 14.3 32 32 32h65V100h-65c-17.7 0-32 14.3-32 32z"
          />
        </svg>
      </span>
      <span
        class="ml-2 hidden md:block text-[0.9rem]"
      >
        UnLikes
      </span>
    </button>
    <button
      aria-label="Navigate to All"
      class="border-b-[3px] bg-transparent decoration-none p-[20px] pb-[5px] mb-[15px] cursor-pointer transition-all duration-150 hover:scale-110 select-none font-bold flex items-center justify-center border-transparent text-white"
      type="button"
    >
      <span
        class="md:hidden"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M487.976 0H24.028C2.71 0-8.047 25.866 7.058 40.971L192 225.941V432c0 7.831 3.821 15.17 10.237 19.662l80 55.98C298.02 518.69 320 507.493 320 487.98V225.941l184.947-184.97C520.021 25.896 509.338 0 487.976 0z"
          />
        </svg>
      </span>
      <span
        class="ml-2 hidden md:block text-[0.9rem]"
      >
        All
      </span>
    </button>
    <button
      aria-label="Navigate to Configs"
      class="border-b-[3px] bg-transparent decoration-none p-[20px] pb-[5px] mb-[15px] cursor-pointer transition-all duration-150 hover:scale-110 select-none font-bold flex items-center justify-center border-transparent text-white"
      type="button"
    >
      <span
        class="md:hidden"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.405 1.05c-.413-1.4-2.397-1.4-2.81 0l-.1.34a1.464 1.464 0 0 1-2.105.872l-.31-.17c-1.283-.698-2.686.705-1.987 1.987l.169.311c.446.82.023 1.841-.872 2.105l-.34.1c-1.4.413-1.4 2.397 0 2.81l.34.1a1.464 1.464 0 0 1 .872 2.105l-.17.31c-.698 1.283.705 2.686 1.987 1.987l.311-.169a1.464 1.464 0 0 1 2.105.872l.1.34c.413 1.4 2.397 1.4 2.81 0l.1-.34a1.464 1.464 0 0 1 2.105-.872l.31.17c1.283.698 2.686-.705 1.987-1.987l-.169-.311a1.464 1.464 0 0 1 .872-2.105l.34-.1c1.4-.413 1.4-2.397 0-2.81l-.34-.1a1.464 1.464 0 0 1-.872-2.105l.17-.31c.698-1.283-.705-2.686-1.987-1.987l-.311.169a1.464 1.464 0 0 1-2.105-.872l-.1-.34zM8 10.93a2.929 2.929 0 1 1 0-5.86 2.929 2.929 0 0 1 0 5.858z"
          />
        </svg>
      </span>
      <span
        class="ml-2 hidden md:block text-[0.9rem]"
      >
        Configs
      </span>
    </button>
  </nav>
</div>
`;
